libhb/scan.c

```diff
@@ -218,6 +221,7 @@ hb_thread_t * hb_scan_init( hb_handle_t * handle, volatile int * die,
     
     data->crop_threshold_frames = crop_threshold_frames;
     data->crop_threshold_pixels = crop_threshold_pixels;
+    data->exclude_extensions = exclude_extensions;
```

Comment:
hb_scan_init will launch a different thread, it might be a good idea to make a copy of exclude_extensions here in case the caller frees it before the new threads has a chance to run and use it.

Reply:
It looks like data is created on each call to hb_scan_init so should be decoupled already unless I#m missing something?

Reply:
It is, but probably it works only because it's never freed (hb_list_close is not called) and it leaks, if you try to free it after calling hb_scan it will crash maybe.
Usually for heap allocated string and struct we make a copy so it doesn't depend on the caller and we can safely use on different threads.
Unfortunately this is C and there is no ready-made way to do a deep copy of the list.

Reply:
You need something like this: https://gist.github.com/galad87/e87dffac4a952996ce0b1fbd1c460d62
And then you call call hb_list_close() after hb_scan in win/CS/HandBrake.Interop/Interop/HandBrakeInstance.cs too.